integration_tests/run_e2e_tests.py

```diff
@@ -360,13 +361,25 @@ def e2e_tests(
         ]
         test_results.extend(results)
 
+    if "adapter_macro_has_default" in test_types:
```

Comment:
Maybe we should split the tests into 2 parts?
The first part will be to run tests like that so we could fail fast, and the second one will be to test the e2e flow.

when thinking about it out-loud this is a unit test so it does make sense to fail fast on it 🙂

Reply:
So it is still part of the e2e test?